src/core/dbus-execute.c

```diff
@@ -2553,6 +2590,60 @@ int bus_exec_context_set_transient_property(
                         return 1;
                 }
 
+        } else if (streq(name, "PathImages")) {
+                const char *source, *destination, *mount_flags;
+                int ignore_enoent;
+                bool empty = true;
+
+                r = sd_bus_message_enter_container(message, 'a', "(ssbs)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(ssbs)", &source, &destination, &ignore_enoent, &mount_flags)) > 0) {
+
+                        if (!path_is_absolute(source))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Source path %s is not absolute.", source);
+                        if (!path_is_absolute(destination))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Destination path %s is not absolute.", destination);
+                        if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                                r = path_image_add(&c->path_images, &c->n_path_images,
+                                                   &(PathImage) {
+                                                           .source = strdup(source),
+                                                           .destination = strdup(destination),
+                                                           .ignore_enoent = ignore_enoent,
+                                                           .mount_flags = strdup(mount_flags),
```

Comment:
oom

Reply:
(passing of ownership is weird here...)

Reply:
This was copy-pasted from BindPaths handling above, so they suffer from same problems.

Reply:
#14466

Reply:
Similarly to what I mentioned in your PR #14466 , no need to do a `strdup`, since your `path_image_add` you copy-pasted from `bind_mount_add` already does a `strdup` of both `source` and `destination`.

Reply:
This was buggy but now fixed.